tests/php/View/AttributesHTMLTest.php

```diff
@@ -0,0 +1,290 @@
+        $dummy->setAttribute('alt', '');
+        $dummy->setAttribute('array', ['foo' => 'bar']);
+        $dummy->setAttribute('width', 123);
+        $dummy->setAttribute('hack>', '">hack');
+
+        $html = $dummy->getAttributesHTML();
+
+        $this->assertStringNotContainsString(
+            'emptystring',
+            $html,
+            'Attribute with empty string are not rendered'
+        );
+
+        $this->assertStringNotContainsString(
+            'nullvalue',
+            $html,
+            'Attribute with null are not rendered'
+        );
+
+        $this->assertStringNotContainsString(
+            'false',
+            $html,
+            'Attribute with false are not rendered'
+        );
+
+        $this->assertStringNotContainsString(
+            'emptyarray',
+            $html,
+            'Attribute with empty array are not rendered'
+        );
+
+        $this->assertStringContainsString(
+            'zeroint="0"',
+            $html,
+            'Attribute with a zero int value are rendered'
+        );
+
+        $this->assertStringContainsString(
+            'zerostring="0"',
+            $html,
+            'Attribute with a zerostring value are rendered'
+        );
+
+        $this->assertStringContainsString(
+            'alt=""',
+            $html,
+            'alt attribute is rendered even when empty set to an empty string'
+        );
+
+        $this->assertStringContainsString(
+            'array="{&quot;foo&quot;:&quot;bar&quot;}"',
+            $html,
+            'Array attribute is converted to JSON'
+        );
+
+        $this->assertStringContainsString(
+            'width="123"',
+            $html,
+            'Numeric values are rendered with quotes'
+        );
+
+        $this->assertStringNotContainsString(
+            'hack&quot;&gt;="&quot;&gt;hack"',
+            $html,
+            'Attribute names and value are escaped'
+        );
+
+        $html = $dummy->getAttributesHTML('zeroint', 'array');
+
+        $this->assertStringNotContainsString(
+            'zeroint="0"',
+            $html,
+            'Excluded attributes are not rendered'
+        );
+
+        $this->assertStringContainsString(
+            'zerostring="0"',
+            $html,
+            'Attribute not excluded still render'
+        );
+
+        $this->assertStringContainsString(
+            'alt=""',
+            $html,
+            'Attribute not excluded still render'
+        );
+
+        $this->assertStringNotContainsString(
+            'array',
+            $html,
+            'Excluded attributes are not rendered'
+        );
+    }
+
+    public function testAttributesHTMLwithExplicitAttr(): void
+    {
+        $dummy = new AttributesHTMLTest\DummyAttributesHTML();
+
+        $this->assertEmpty(
+            '',
+            $dummy->getAttributesHTML(),
+            'If no attributes are provided, an empty string should be returned'
+        );
+
+        $attributes = [
```

Comment:
Use a data provider for this instead

Reply:
`getAttributesHTML` takes all the attributes and combined them into one string.

I'm not trying to test individuals scenarios here. I'm trying to test the scenario where you combined all these attributes together.